### PR TITLE
shotgun shell clips

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
Shotgun shell clips. Small items, hold 4 shells in 1 slot, can be used to reload a shotgun (one shell per click). Not roundstart - you'll have to figure out that they exist, first, before you can get your grubby mitts on them.
Credits to BobJoga for the original code and sprites that are being pasted here under the usual AGPL and CC-BY-SA licenses.
## Why It's Good For The Game
Shotgun shells are a bit of a hassle to throw around. This might make things a little better if you have to lug around boxes of shells - then again, if you're lugging around boxes of shells, you might have more pressing matters.
## Changelog
:cl: BobJoga (ported by Hatterhat)
add: Rumors of the existence of metal strips designed to hold shotgun shells have reached Citadel Station 13.
/:cl:

### CODE REVIEW

**Comment by BobJogaIsForkinBanned** [comment]
oh hi

**Comment by ghost** [comment]
(Removed after credit was given, ty Ghommie)
We are very proud of our favorite content creator who breaks things all the time. Make sure you test this properly :3
https://github.com/Skyrat-SS13/Skyrat13/pull/1405

**Comment by Strazyplus** [comment]
Big brain

**Comment by ghost** [comment]
(Removed after credit was given, ty Ghommie)

**Comment by Draganfrukts** [comment]
Hey, wait, I know this one!

**Comment by Ghommie** [comment]
DNM tag until the original PR is merged downstream (even though it there would be no issue since it's made using pre-existing AGPL/CC-BY-SA code and assets, this is more of a cordiality)

**Comment by Putnam3145** [comment]
hell's going on in the comments in here

**Comment by Hatterhat** [comment]
looks like i owe myself five bucks for not getting away with that

**Comment by Draganfrukts** [comment]
Dismantled and bluepilled

**Comment by ShadeAware** [comment]
TF is wrong with you Hatter smh

**Comment by ghost** [comment]
@Hatterhat Credit us next time or we will show up at your door with soap socks 🔫

**Comment by Hatterhat** [comment]
@CinderWC square up nerd

**Comment by Putnam3145** [comment]
@CinderWC okay while i do think it's good practice to credit people let it be clear that it is neither in the spirit nor the letter of our licenses to do that

**Comment by ghost** [comment]
@Putnam3145 Oh i never meant any of this in a law sense as in "Ima sick the lawyers onto you"
I just wanted us/our contributors to be credited for this as its just in bad spirit to take without credit whatsoever.

**Comment by MrJWhit** [comment]
I don't really see it as bad spirit, take what you want or not. Credit is nice, but not required in any way.

**Comment by BobJogaIsForkinBanned** [comment]
> I don't really see it as bad spirit, take what you want or not. Credit is nice, but not required in any way.

Credit isn't required - but it's appreciated. Copying something without at least crediting the original is bad spirit, but i wouldn't say it's legally worrisome.

**Comment by Hatterhat** [comment]
copying something without crediting the original sounds like what downstreams do :^)

**Comment by CinderWC** [comment]
> 
> 
> copying something without crediting the original sounds like what downstreams do :^)

Sounds like it, but in theory it seems to be Contributors on upstream that seem to really like to do it. Case in point.

**Comment by Hatterhat** [comment]
turnabout is fair play

**Comment by CinderWC** [comment]
5ef42524816b9b7da110641149508307

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 1]
```diff
@@ -121,3 +121,55 @@
...
+	var/pixeloffsetx = 4
+	var/shell_overlay_list = list(null, null, null, null)
+
+/obj/item/ammo_box/shotgun/update_icon()
+	..()
+	update_overlays()
+
+/obj/item/ammo_box/shotgun/update_overlays()
+	. = ..()
+	cut_overlay(shell_overlay_list, TRUE)
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]: Remove.

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 2]
```diff
@@ -121,3 +121,55 @@
...
+	icon = 'icons/obj/ammo.dmi'
+	icon_state = "shotgunclip"
+	max_ammo = 4
+	ammo_type = /obj/item/ammo_casing/shotgun
+	var/pixeloffsetx = 4
+	var/shell_overlay_list = list(null, null, null, null)
+
+/obj/item/ammo_box/shotgun/update_icon()
+	..()
+	update_overlays()
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]: This is already done, remove.

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 3]
```diff
@@ -121,3 +121,55 @@
...
+		var/offset = -4
+		for(var/obj/item/ammo_casing/shotgun/C in stored_ammo)
+			shellindex++
+			offset += pixeloffsetx
+			var/mutable_appearance/shell_overlay = mutable_appearance(icon, "[initial(C.icon_state)]-clip")
+			shell_overlay.pixel_x += offset
+			shell_overlay.appearance_flags = RESET_COLOR
+			shell_overlay_list[shellindex] = shell_overlay
+			C.current_overlay = shell_overlay
+			add_overlay(shell_overlay, TRUE)
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]:
```suggestion
			. += shell_overlay
```

File: code/modules/projectiles/ammunition/ballistic/shotgun.dm [hunk 4]
```diff
@@ -7,6 +7,7 @@
 	caliber = "shotgun"
 	projectile_type = /obj/item/projectile/bullet/shotgun_slug
 	custom_materials = list(/datum/material/iron=4000)
+	var/mutable_appearance/current_overlay // for stripper clips
```
Comment (Ghommie) [inline code/modules/projectiles/ammunition/ballistic/shotgun.dm]: pretty much unused, remove.

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 5]
```diff
@@ -121,3 +121,55 @@
...
+		var/shellindex = 0
+		var/offset = -4
+		for(var/obj/item/ammo_casing/shotgun/C in stored_ammo)
+			shellindex++
+			offset += pixeloffsetx
+			var/mutable_appearance/shell_overlay = mutable_appearance(icon, "[initial(C.icon_state)]-clip")
+			shell_overlay.pixel_x += offset
+			shell_overlay.appearance_flags = RESET_COLOR
+			shell_overlay_list[shellindex] = shell_overlay
+			C.current_overlay = shell_overlay
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]: Idem.

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 6]
```diff
@@ -121,3 +121,55 @@
...
+//Shotgun clips
+/obj/item/ammo_box/shotgun
+	name = "stripper clip (shotgun shells)"
+	desc = "A stripper clip, designed to help with loading a shotgun slightly faster."
+	icon = 'icons/obj/ammo.dmi'
+	icon_state = "shotgunclip"
+	max_ammo = 4
+	ammo_type = /obj/item/ammo_casing/shotgun
+	var/pixeloffsetx = 4
+	var/shell_overlay_list = list(null, null, null, null)
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]: remove.

File: code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 7]
```diff
@@ -121,3 +121,55 @@
...
+	..()
+	update_overlays()
+
+/obj/item/ammo_box/shotgun/update_overlays()
+	. = ..()
+	cut_overlay(shell_overlay_list, TRUE)
+	if(stored_ammo.len)
+		var/shellindex = 0
+		var/offset = -4
+		for(var/obj/item/ammo_casing/shotgun/C in stored_ammo)
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/ammo_boxes.dm]:
```suggestion
		for(var/A in stored_ammo)
			var/obj/item/ammo_casing/shotgun/C = A
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Useroth & co told me to just merge and spare them from modular conflicts, but there are some glaring hacky code that has to be dealt with. See #11976.

**Comment by silicons** [comment]
I'm not exactly a fan of buffing shotguns this way.
They're meant to be clunky.
That plus and this'll need inventory stuff done considering they currently take the same amount of space as just a regular shell (guess throwing normal sized on it would work though, volumetric system automatically handles it and it'd be less efficient than boxes as a trade-off).
@Citadel-Station-13/project-management

**Review by Ghommie — APPROVED** [review]
To be honest, these are just shell holders, not stripper clips, and probably are gonna stay as such here.

**Pull request merged**